cryptographic_flows.md

```diff
@@ -22,8 +22,36 @@ An asset owner that has not previously interacted with the key server MUST regis
     1. Checks that `user_id` matches that of the authenticated user in the open registration session. If this check fails, the server MUST abort.
     1. Checks that the received ciphertext is of the expected format and length.
     1. [Stores](#server-side-storage) the received ciphertext in a record associated with `user_id`.
+    1. Stores the registration registration request in the [audit log](#audit-logs).
 1. The client and server close the session upon completion of the request.
 
+## Retrieve Audit Logs
+The asset owner can request audit logs from the key server.
+
+Input:
+- `user_id`, a 128-bit globally unique identifier (GUID) representing the identity of the asset owner.
+- `type`, one of:
+    - `"system only"`, which indicates the asset owner wants a record of registration and login actions;
+    - `"key only"`, which indicates the asset owner wants a record of requested key use operations with respect to or more keys; or
+    - `"all"`, which indicates the asset owner wants both system and requested key use operations.
+- `key_identifiers`, an OPTIONAL list of key identifiers.
+
+Output: `summary_record`, which contains the requested history, including timestamps.
+
+Protocol:
+
+1. The client:
+   1. [Opens a request session](systems-architecture.md#request-session) for the given user id `user_id`. 
+   1. Sends a request message to the key server over the session's secure channel. This message MUST indicate the desire to retrieve audit logs, specify the type of audit requested (i.e., `"system only"`, `"key only"`, or `"all"`) and contain `user_id`.
+1. The key server:
+   1. Runs a validity check on the received request and `user_id`(i.e., there must be a valid open request session, the request must conform to the expected format, and `user_id` must be of the expected format and length, and should match that of the open request session). If this check fails, the server MUST reject the request.
+   1. Retrieves all log records relevant to the client's request, summarises these records in `summary_record`, and sends `summary_record` to the client over the secure channel.
+   1. Stores the current request information, including the outcome of the validity check, in an [audit log](#audit-logs) associated with the given user.    
```

Comment:
Does this mean we make a log entry for someone having requested the logs? Also does this type of log fall under "system log"?

Reply:
Yes, and yes.